### PR TITLE
update the challenge

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

**Review by Shwetabhk — APPROVED** [review]

File: core/models/assignments.py
Comment (Anugnareddymule) [inline core/models/assignments.py:1]: class ClassName:
    @classmethod
    def get_assignments_by_teacher_id(cls, teacher_id):
        return cls.query.filter(cls.teacher_id == teacher_id).all()

    @classmethod
    def get_all_assignments(cls):
        return cls.query.all()
in 92 line

File: tests/SQL/sql_test.py
Comment (Anugnareddymule) [inline tests/SQL/sql_test.py:1]: import unittest
import sqlite3

class TestSQLQueries(unittest.TestCase):
    def setUp(self):
        # Set up a connection to a test database
        self.conn = sqlite3.connect(':memory:')
        self.cur = self.conn.cursor()

        # Create a test table with sample data
        self.cur.execute('''CREATE TABLE assignments (
                                id INTEGER PRIMARY KEY,
                                state TEXT
                            )''')
        self.cur.execute('''INSERT INTO assignments (state) VALUES
                            ('State1'),
                            ('State2'),
                            ('State1'),
                            ('State3'),
                            ('State2'),
                            ('State2')''')

    def tearDown(self):
        # Clean up by closing the database connection
        self.conn.close()

    def test_count_assignments_by_state(self):
        # Arrange: Prepare the query
        query = '''SELECT state, COUNT(*) AS num_assignments
                   FROM assignments
                   GROUP BY state'''
        
        # Act: Execute the query
        self.cur.execute(query)
        result = self.cur.fetchall()

        # Assert: Verify the results
        expected_result = [('State1', 2), ('State2', 3), ('State3', 1)]
        self.assertEqual(result, expected_result)

if __name__ == '__main__':
    unittest.main()

File: core/apis/assignments/teacher.py
Comment (Anugnareddymule) [inline core/apis/assignments/teacher.py:1]: from flask import Blueprint, jsonify
from core.models import Assignment

teacher_api = Blueprint('teacher_api', __name__)

@teacher_api.route('/assignments/teacher/<int:teacher_id>/states', methods=['GET'])
def get_assignments_by_state(teacher_id):
    # Assuming Assignment model has a column named 'state'
    assignments_by_state = Assignment.get_assignments_by_teacher_id(teacher_id)
    
    # Count the number of assignments for each state
    state_counts = {}
    for assignment in assignments_by_state:
        state = assignment.state
        state_counts[state] = state_counts.get(state, 0) + 1
    
    return jsonify(state_counts), 200

File: Application.md
Comment (Anugnareddymule) [inline Application.md:1]: # Application Name

## Overview
[Provide a brief overview of your application. Describe its purpose and target audience.]

## Features
- [Feature 1]
- [Feature 2]
- [Feature 3]
- ...

## Technologies Used
- [Technology 1]
- [Technology 2]
- [Technology 3]
- ...

## Setup Instructions
[Provide instructions on how to set up and run your application locally.]

1. Clone the repository:
   ```bash
   git clone https://github.com/your-username/your-repository.git
cd your-repository
pip install -r requirements.txt
run the code:
python app.py

**Pull request merged**